velox/exec/Spiller.h

```diff
@@ -86,6 +99,10 @@ class Spiller {
       memory::MemoryPool* pool,
       folly::Executor* executor);
```

Comment:
Is it possible to consolidate these 4 constructors? It looks like they are referring back to the same "base" ctor. We can have a base one that has all needed parameters and have a couple of parameters optional?

Reply:
We might revisit this like either have a derived class for each different spiller type. Have chat with @oerling, we can refactor this after having implemented all the use cases. Like spilling support in hash probe stage.